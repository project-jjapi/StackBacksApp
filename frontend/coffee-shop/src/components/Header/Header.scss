.Header {
  width: 80%;
  margin: 0 auto;
  display: flex;
  justify-content: flex-end;
  align-items: center;
  // box-shadow: 0 5px 10px 0 rgba(0, 64, 128, 0.05);
  & figure {
    margin: 0;
    margin-right: auto;
  }
  &-menu {
    align-content: center;
    & ul {
      list-style: none;
      justify-content: space-between;
      & li {
        display: inline-block;
        padding: 0px 20px;
        font-weight: 600;
        text-transform: uppercase;
        transition: all 0.3s ease 0s;
        &:hover {
          color: #0088a9;
        }
      }
    }
  }
  &-button {
    margin: 0 30px;
    padding: 9px 25px;
    background-color: rgba(0, 136, 169, 0.9);
    border: none;
    color: white;
    border-radius: 50px;
    cursor: pointer;
    outline: none;
    transition: all 0.3s ease 0s;
  }
}
