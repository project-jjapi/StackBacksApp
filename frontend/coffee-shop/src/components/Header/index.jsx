import React from 'react'
import { useDispatch } from 'react-redux'

import { showModalAction } from '../../redux/actions'

import LogoCoffeeShop from '../../img/cover.png'

import './Header.scss'

export const Header = () => {
  const dispatch = useDispatch()

  const showModal = () => {
    dispatch(showModalAction(true))
  }

  return (
    <header className='Header'>
      <figure>
        <img src={LogoCoffeeShop} width='180px' alt='logo' />
      </figure>
      <nav className='Header-menu'>
        <ul>
          <li>Inicio</li>
          <li>Menú</li>
          <li>Tiendas</li>
          <li>Conócenos</li>
        </ul>
      </nav>
      <div className='Header-user'>
        <button className='Header-button' onClick={() => showModal()}>
          Iniciar sesión
        </button>
      </div>
    </header>
  )
}

export default Header
