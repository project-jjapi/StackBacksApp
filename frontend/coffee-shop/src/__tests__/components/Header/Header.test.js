import React from 'react'
import { mount } from 'enzyme'
import { create } from 'react-test-renderer'

/**
 * import component Header
 */
import Header from '../../../components/Header'

/**
 * TODO: Test component header
 */

describe('<Header />', () => {
  const wrapper = mount(<Header />)
  test('should render one header', () => {
    expect(wrapper.length).toEqual(1)
  })
  test('should have four links', () => {
    expect(wrapper.find('li')).toHaveLength(4)
  })
  test('should create snapshot', () => {
    const snapshot = create(<Header />)
    expect(snapshot.toJSON()).toMatchSnapshot()
  })
})
