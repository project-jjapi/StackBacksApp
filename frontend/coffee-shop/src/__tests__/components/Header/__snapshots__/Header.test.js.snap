// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should create snapshot 1`] = `
<div
  className="Header"
>
  <figure>
    <img
      alt="logo"
      src="cover.png"
      width="180px"
    />
  </figure>
  <div
    className="Header-menu"
  >
    <ul>
      <li>
        Inicio
      </li>
      <li>
        Menú
      </li>
      <li>
        Tiendas
      </li>
      <li>
        Conócenos
      </li>
    </ul>
  </div>
  <div
    className="Header-user"
  >
    <button
      className="Header-button"
    >
      Iniciar sesión
    </button>
  </div>
</div>
`;
